hosted/gke/p1/p1_provisioning_test.go

```diff
@@ -306,4 +313,71 @@ var _ = Describe("P1Provisioning", func() {
 			updateClusterInUpdatingState(cluster, ctx.RancherAdminClient)
 		})
 	})
+
+	FWhen("a private cluster is created", func() {
+
+		var err error
+		var updateFunc func(clusterConfig *gke.ClusterConfig)
+		network := pointer.String("hosted-providers-ci-private")
+
+		BeforeEach(func() {
+			updateFunc = func(clusterConfig *gke.ClusterConfig) {
+				clusterConfig.Network = network
+				clusterConfig.Subnetwork = network
+				clusterConfig.PrivateClusterConfig.EnablePrivateNodes = true
+				clusterConfig.PrivateClusterConfig.MasterIpv4CidrBlock = fmt.Sprintf("172.16.%d.0/28", rand.Intn(10))
+
+				currentSpec := CurrentSpecReport().FullText()
+				if strings.Contains(currentSpec, "MasterAuthorizedNetworks") {
+					_, authorizedIP, err := net.ParseCIDR(helpers.GetRancherIP() + "/32")
+					Expect(err).To(BeNil())
+
+					newCidrBlock := gke.CidrBlock{
+						CidrBlock:   authorizedIP.String(),
+						DisplayName: "Rancher",
+					}
+					clusterConfig.MasterAuthorizedNetworksConfig.Enabled = true
+					clusterConfig.MasterAuthorizedNetworksConfig.CidrBlocks = append(clusterConfig.MasterAuthorizedNetworksConfig.CidrBlocks, newCidrBlock)
+				}
+			}
+
+			cluster, err = helper.CreateGKEHostedCluster(ctx.RancherAdminClient, clusterName, ctx.CloudCredID, k8sVersion, zone, project, updateFunc)
+			Expect(err).To(BeNil())
+
+			cluster, err = helpers.WaitUntilClusterIsReady(cluster, ctx.RancherAdminClient)
+			Expect(err).To(BeNil())
+
+			Expect(cluster.GKEConfig.PrivateClusterConfig.EnablePrivateNodes).To(BeTrue())
+		})
+
+		It("should successfully create with public endpoint", func() {
+			testCaseID = 22
+
+			Eventually(func() bool {
+				GinkgoLogr.Info("Waiting for the private config to appear in GKEStatus.UpstreamSpec...")
+				cluster, err := ctx.RancherAdminClient.Management.Cluster.ByID(cluster.ID)
+				Expect(err).To(BeNil())
+				return cluster.GKEStatus.UpstreamSpec.PrivateClusterConfig.EnablePrivateNodes
+			}, "2m", "15s").Should(BeTrue(), "Failed while waiting for private config to appear")
+
+			cluster, err = helper.AddNodePool(cluster, ctx.RancherAdminClient, 1, "", true, true)
+			Expect(err).To(BeNil())
+		})
+
+		It("should successfully create with public endpoint and MasterAuthorizedNetworks", func() {
+			testCaseID = 24
+
+			Expect(cluster.GKEConfig.MasterAuthorizedNetworksConfig.Enabled).To(BeTrue())
+			Eventually(func() bool {
```

Comment:
Do we need to wait for upstream spec? When the cluster is first created, UpstreamSpec and ClusterConfig must have the same configuration.

Reply:
Right,  UpstreamSpec and ClusterConfig can be checked simultaneously